pkg/antctl/command_definition.go

```diff
@@ -480,6 +495,136 @@ func (cd *commandDefinition) tableOutputForGetCommands(obj interface{}, writer i
 	return nil
 }
 
+
+// tableOutputForQueryEndpoint implements printing sub tables (list of tables) for each response, utilizing constructTable
+// with multiplicity.
+func (cd *commandDefinition) tableOutputForQueryEndpoint(obj interface{}, writer io.Writer) error {
+	// intermittent new line buffer
+	var buffer bytes.Buffer
+	newLine := func() error {
+		buffer.WriteString("\n")
+		if _, err := io.Copy(writer, &buffer); err != nil {
+			return fmt.Errorf("error when copy output into writer: %w", err)
+		}
+		buffer.Reset()
+		return nil
+	}
+	// sort rows of sub table
+	sortRows := func(rows [][]string) {
+		body := rows[1:]
+		sort.Slice(body, func(i, j int) bool {
+			for k := range body[i] {
+				if body[i][k] != body[j][k] {
+					return body[i][k] < body[j][k]
+				}
+			}
+			return true
+		})
+	}
+	// constructs sub tables for responses
+	constructSubTable := func(header [][]string, body [][]string) error {
+		rows := append(header, body...)
+		sortRows(rows)
+		numRows, numCol := len(rows), len(rows[0])
+		widths := getColumnWidths(numRows, numCol, rows)
+		if err := constructTable(numRows, numCol, widths, rows, writer); err != nil {
+			return err
+		}
+		return nil
+	}
+	// construct sections of sub tables for responses (applied, ingress, egress)
+	constructSection := func(label [][]string, header [][]string, body [][]string, nonZero bool) error {
+		if err := constructSubTable(label, [][]string{}); err != nil {
+			return err
+		}
+		if nonZero {
+			if err := constructSubTable(header, body); err != nil {
+				return err
+			}
+		}
+		if err := newLine(); err != nil {
+			return err
+		}
+		return nil
+	}
+	// response is string representation of endpoint response
+	type response struct {
+		Namespace    string
+		Name         string
+		Policies     [][]string
+		EgressRules  [][]string
+		IngressRules [][]string
+	}
+	// iterate through each endpoint and construct response
+	endpointQueryResponse := obj.(*networkpolicy.EndpointQueryResponse)
+	responses := make([]*response, 0)
+	for _, endpoint := range endpointQueryResponse.Endpoints {
```

Comment:
I don't think you need `response` here? you could work on `EndpointQueryResponse` directly and just build `policies`, `ingressRules`, `egressRules` which correspond to our different sections.